tutorials/terraform-asm-in-cluster.md

```diff
@@ -0,0 +1,445 @@
+        apiVersion: install.istio.io/v1alpha1
+        kind: IstioOperator
+        spec:
+          components:
+            ingressGateways:
+            - name: istio-ingressgateway
+              enabled: true
+              k8s:
+                hpaSpec:
+                  maxReplicas: 10
+                  minReplicas: 2
+        EOF
+
+## Deploy resources with Terraform
+
+In this section, you create and apply Terraform files that define the deployment of a VPC network, GKE cluster, and Anthos Service Mesh.
+
+1.  Create the `main.tf`, `variables.tf`, and `output.tf` files:
+
+        cat <<'EOF' > main.tf_tmpl
+        data "google_client_config" "default" {}
+
+        provider "kubernetes" {
+          host                   = "https://${module.gke.endpoint}"
+          token                  = data.google_client_config.default.access_token
+          cluster_ca_certificate = base64decode(module.gke.ca_certificate)
+        }
+
+        data "google_project" "project" {
+          project_id = var.project_id
+        }
+
+        module "vpc" {
+          source  = "terraform-google-modules/network/google"
+          version = "~> 3.0"
+
+          project_id   = var.project_id
+          network_name = var.network
+          routing_mode = "GLOBAL"
+
+          subnets = [
+            {
+              subnet_name   = var.subnetwork
+              subnet_ip     = var.subnetwork_ip_range
+              subnet_region = var.region
+            }
+          ]
+
+          secondary_ranges = {
+            (var.subnetwork) = [
+              {
+                range_name    = var.ip_range_pods
+                ip_cidr_range = var.ip_range_pods_cidr
+              },
+              {
+                range_name    = var.ip_range_services
+                ip_cidr_range = var.ip_range_services_cidr
+              }
+            ]
+          }
+        }
+
+        module "gke" {
+          source                  = "terraform-google-modules/kubernetes-engine/google"
+          project_id              = var.project_id
+          name                    = var.cluster_name
+          regional                = false
+          region                  = var.region
+          zones                   = var.zones
+          release_channel         = "REGULAR"
+          network                 = module.vpc.network_name
+          subnetwork              = module.vpc.subnets_names[0]
+          ip_range_pods           = var.ip_range_pods
+          ip_range_services       = var.ip_range_services
+          network_policy          = false
+          identity_namespace      = "enabled"
+          cluster_resource_labels = { "mesh_id" : "proj-${data.google_project.project.number}" }
+          node_pools = [
+            {
+              name         = "asm-node-pool"
+              autoscaling  = false
+              auto_upgrade = true
+              # ASM requires minimum 4 nodes and e2-standard-4
+              node_count   = 2
```

Comment:
The comment seems to contradict the node_count setting.

This seeming discrepancy exists in the other documents, too.

Reply:
Updated all docs to node_count = 4